File: 199-binary-tree-right-side-view/binary-tree-right-side-view.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        def SideViewHelper(root, rightSideArr, level):
            if level == len(rightSideArr):
                rightSideArr.append(root.val)
            
            for child in [root.right, root.left]:
                if child:
                    SideViewHelper(child, rightSideArr, level + 1)
        
        if not root: return []
        
        rightSideArr = []
        SideViewHelper(root, rightSideArr, 0)
        
        return rightSideArr

File: 199-binary-tree-right-side-view/binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        if (root == null) {
            return new ArrayList<>();
        }

        List<Integer> rSide = new ArrayList<>();
        this.dfs(root, rSide, 0);
        return rSide; 
        
    }

    public void dfs(TreeNode root, List<Integer> rSide, int level) {
        if (root == null) {
            return ;
        }

        if(level == rSide.size()) {
            rSide.add(root.val);
        }

        this.dfs(root.right, rSide, level + 1);
        this.dfs(root.left, rSide, level + 1);   
    }
}